File: src/dataset/javascript/matrix_exercism-ThomasZumsteg/claude/ClaudeSonnet4_matrix_v4.js
var Matrix = function(matrix) {
	var splitRows = matrix.split("\n");
	var numRows = splitRows.length;
	var numCols = splitRows[0].split(" ").length;
	
	this.rows = new Array(numRows);
	for (var i = 0; i < numRows; i++) {
		var rowElements = splitRows[i].split(" ");
		this.rows[i] = new Array(numCols);
		for (var j = 0; j < numCols; j++) {
			this.rows[i][j] = parseInt(rowElements[j], 10);
		}
	}

	this.columns = new Array(numCols);
	for (var col = 0; col < numCols; col++) {
		this.columns[col] = new Array(numRows);
		for (var row = 0; row < numRows; row++) {
			this.columns[col][row] = this.rows[row][col];
		}
	}
};

module.exports = Matrix;

File: src/dataset/python/matrix_exercism-ThomasZumsteg/claude/ClaudeSonnet4_matrix_v4.py
"""A 2 dimentional matrix object"""

class Matrix(object):
    """Creates a 2d matrix object with accessable rows and columns"""

    def __init__(self, matrix):
        """Copy of the input data"""
        self.matrix = matrix
        self._rows = None
        self._columns = None

    @property
    def rows(self):
        """Matrix displayed as rows, coumns"""
        if self._rows is None:
            self._rows = [list(map(int, row.split())) 
                         for row in self.matrix.split("\n")]
        return self._rows

    @property
    def columns(self):
        """Matrix displayed as columns, rows"""
        if self._columns is None:
            rows = self.rows
            self._columns = list(map(list, zip(*rows)))
        return self._columns